### PR TITLE
Retargeting patch series.

### PR DESCRIPTION
**Pull request by eric-s-raymond**
It's done. Ready to be reviewed and landed,

### CODE REVIEW

File: src/main.c [hunk 1]
```diff
@@ -1797,10 +1709,10 @@ void usage (void)
 {
 	FILE   *f = stdout;
 
-	if (!did_outfilename) {
+	if (!env.did_outfilename) {
 		snprintf (outfile_path, sizeof(outfile_path), outfile_template,
-			 prefix, C_plus_plus ? "cc" : "c");
-		outfilename = outfile_path;
+			  ctrl.prefix, backend->suffix());
```
Comment (db48x) [inline src/main.c:1714]: `backend` is still null at this point.

Comment (westes) [reply src/main.c]: @eric-s-raymond did you address this?

**Comment by Mightyjo** [comment]
I'm reviewing this PR and getting an error during make dist and make distcheck.
```
make[4]: *** No rule to make target 'test-yydecl-c99.c', needed by 'distdir-am'.  Stop.
make[4]: Leaving directory '/mnt/d/Users/Joe/Documents/GitHub/flex/tests'
```
Q1: Do you get this error when building from a clean source tree? (git clean -xf; ./autogen.sh && ./configure && make && make distcheck) Could be I'm configured oddly and need to fix myself.

Q2: I see that test-yydecl-* are built by test-yydecl-gen.sh. Am I correct in thinking the intent is these will be regenerated from the distribution tarball? If you're getting the same error, I think we can generate some intermediate rules to satisfy make.

**Comment by westes** [comment]
Yes, I get that same error when building from a clean source tree.

The issue is that there are no rules to tell automake about these entries in the test suite. That's what needs fixed and I have just not been able to find a slot of time to work on it.

**Comment by Mightyjo** [comment]
Copy that. Working on it.

**Comment by Mightyjo** [comment]
Got 'make dist' calmed down by telling it to stop creating default _SOURCES variables for the test-yydecl scripts. Sent a PR to ESR's branch. Link to the commit [here: e50f889](https://github.com/Mightyjo/flex/commit/e50f889082e8a5ee301222b0495ad3d6aa312506).

**Comment by Mightyjo** [comment]
Exposed a new build order quirk regarding the headers produced from the skeletons. Working on that now.

**Comment by Mightyjo** [comment]
Sent a few more patches over in the PR to ESR. 

src/ and doc/ are building under distcheck. The testmaker.sh script needs a hint about VPATH, but that's for tomorrow Joe.

**Comment by Mightyjo** [comment]
Tomorrow Joe got it done. The PR I sent to ESR is passing distcheck. Had to bring the test inputs back into the repo instead of generating them on the fly. The VPATH indirection got to be too much for me. Beside which, he'd already done a good job of condensing them down to one text file per class of test cases.

At this point, @eric-s-raymond can merge my PR on his repo or I can make a PR based on his changes (in turn based on HEAD here).

Standing by.

**Comment by westes** [comment]
Please go ahead and submit a pull request against this repository. I think you can just add them to this pull request. If not, then submit a separate pull request and I can get the two pr's onto master.

**Comment by Mightyjo** [comment]
I didn't have write permissions from any PRs to add to this PR. I made a consolidated PR #489.

**Comment by westes** [comment]
Thank you!

**Comment by westes** [comment]
Closing in preference to #489

**Pull request closed without merging**